src/Api/ShippingLabelFetcher.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusDhl24PlShippingExportPlugin\Api;
+
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+
+class ShippingLabelFetcher implements ShippingLabelFetcherInterface
+{
+    /** @var WebClientInterface */
+    private $webClient;
+
+    /** @var SoapClientInterface */
+    private $soapClient;
+
+    /** @var string */
+    private $response;
+
+    private FlashBagInterface $flashBag;
+
+    public function __construct(FlashBagInterface $flashBag, WebClientInterface $webClient, SoapClientInterface $soapClient)
+    {
+        $this->flashBag = $flashBag;
+        $this->webClient = $webClient;
+        $this->soapClient = $soapClient;
+    }
+
+    public function createShipment($shippingGateway, $shipment): void
+    {
+        $this->webClient->setShippingGateway($shippingGateway);
+        $this->webClient->setShipment($shipment);
+
+        try {
+            $requestData = $this->webClient->getRequestData();
+
+            $this->response = $this->soapClient->createShipment($requestData, $shippingGateway->getConfigValue('wsdl'));
+        } catch (\Exception $exception) {
```

Comment:
Shouldn't be so wide? Why not handle a particular exception or provide a list of handled exception here?